src/world/area_mgm/mgm_01/E12930.c

```diff
@@ -1,59 +1,667 @@
+    for(i = 0; i < ARRAY_COUNT(data->panels); i++) {
+        if (blockType == data->panels[i].type && data->panels[i].state == PANEL_STATE_START_ANIM) {
+                data->panels[i].state = PANEL_STATE_EMERGE_INIT;
+                data->panels[i].blockPosIndex = index;
+                data->panels[i].tallyPosIndex = historyPos;
+                break;
+        }
+    }
+
+    scorekeeper->varTable[BROKEN_BLOCKS_VAR_IDX]++;
+    return ApiStatus_DONE2;
+}
+
+ApiStatus N(CreateBlockEntities)(Evt* script, s32 isInitialCall) {
+    JumpGameData* data = get_enemy(SCOREKEEPER_ENEMY_IDX)->varTablePtr[JUMP_DATA_VAR_IDX];
+    s32 entityIndex;
+    s32 initialConfiguration;
+    s32 indexA, indexB;
+    s32 curBlockIdx;
+    s32 temp;
+    s32 i;
+
+    EvtScript* scriptArray[] = {
+        &D_80242310, &D_80242330, &D_80242350, &D_80242370, &D_80242390, &D_802423B0, &D_802423D0, &D_802423F0,
+        &D_80242410, &D_80242430, &D_80242450
+    };
+
+    if (isInitialCall) {
+        // choose one of four initial configurations at random
+        initialConfiguration = rand_int(1000) % ARRAY_COUNT(N(InitialConfigurations));
+        for(i = 0; i < NUM_BLOCKS; i++) {
+            data->type[i] = N(InitialConfigurations)[initialConfiguration][i];
+        }
+
+        // randomly swap 1000 pairs
+        for(i = 0; i < 1000; i++) {
+            indexA = rand_int(10000) % NUM_BLOCKS;
+            indexB = rand_int(10000) % NUM_BLOCKS;
+
+            if (indexA != indexB) {
+                temp = data->type[indexB];
+                data->type[indexB] = data->type[indexA];
+                data->type[indexA] = temp;
+            }
+        }
+        
+        script->functionTemp[0] = 0;
+        script->functionTemp[1] = 0;
+    }
+
+    if (--script->functionTemp[0] <= 0) {
+        curBlockIdx = script->functionTemp[1];
+        entityIndex = create_entity(&D_802EA0C4,
+            N(BlockPosX)[curBlockIdx],
+            N(BlockPosY)[curBlockIdx],
+            N(BlockPosZ)[curBlockIdx],
+            0, 0, 0, 0, MAKE_ENTITY_END);
+        data->panels[curBlockIdx].entityIndex = entityIndex;
+        get_entity_by_index(entityIndex)->boundScriptBytecode = scriptArray[curBlockIdx];
+        fx_sparkles(3,
+            N(BlockPosX)[curBlockIdx],
+            N(BlockPosY)[curBlockIdx] + 13,
+            N(BlockPosZ)[curBlockIdx] + 5,
+            23.0f);
+        sfx_play_sound(SOUND_213);
+        script->functionTemp[0] = 3;
+        script->functionTemp[1]++;
+    }
+
+    return (script->functionTemp[1] < NUM_BLOCKS) ? ApiStatus_BLOCK : ApiStatus_DONE2;
+}
+
+const char* mgm_01_str = "mgm_00";
+
+ApiStatus N(TakeCoinCost)(Evt* script, s32 isInitialCall) {
+    PlayerData* playerData = &gPlayerData;
+
+    if (isInitialCall) {
+        playerData->jumpGamePlays++;
+        script->functionTemp[0] = 0;
+    }
+    add_coins(-1);
+    sfx_play_sound(SOUND_211);
+
+    return (++script->functionTemp[0] == PLAY_COST) ? ApiStatus_DONE2 : ApiStatus_BLOCK;
```

Comment:
script->functionTemp[0]++ on the prev line, followed by checking if it's equal to PLAY_COST plz